builder/src/extensionConfig.ts

```diff
@@ -194,33 +194,49 @@ function generateConfig({
     }
   }
 
+  // Allow custom webpack config
+  let webpackConfigPath = data.jupyterlab['webpackConfig'];
+  let webpackConfig = {};
+
+  // Use the custom webpack config only if the path to the config
+  // is specified in package.json (opt-in)
+  if (webpackConfigPath) {
```

Comment:
This makes it opt-in, meaning that even if a `webpack.config.js` file is at the root of the federated extension folder, it will not be picked up unless explicitly specified in `package.json` under the `jupyterlab.webpackConfig` subkey.

Reply:
Thanks. I think that is really important to be explicit about this, especially given it is an experimental feature. We only want it available to people that really explicitly want to use it.